lib/services/sf.js

```diff
@@ -571,6 +574,12 @@ function StateAbstract(options) {
 
             err = Errors.createNetworkError(
               ErrorCodes.ERR_SF_NETWORK_COULD_NOT_CONNECT, err);
+
+            const auth = getAuthenticator(connectionConfig, httpClient);
+            if (auth instanceof AuthOkta) {
+              Logger.getInstance().debug('OKTA authentication requires token refresh.');
+              await auth.reauthenticate(requestOptions.json);
```

Comment:
Would that approach re-initialize retry timeout and retry count each time and end up infinite retry if the server side keep returning retry-able error code like 429?

Reply:
I tested this as the driver throws an Request error with the wrong query to test the retry strategy instead of the network Error. This approach does not reset the retry count because it just creates the error instance. This instance is  passed to the  requestcallback as a parameter(https://github.com/snowflakedb/snowflake-connector-nodejs/pull/745/files#diff-333982e8b49a2955b892e81f109de2f00423a1cdbe166258fb1eded7a6c906e2R1123) and this function will manage the retry.

As a result, I do not think this makes an infinite loop although the server throws 429 error.

Please let me know if I misunderstand your question.